### PR TITLE
Changed FormatStrFormatter documentation to include how to get unicode minus

### LINKED ISSUE #17967: Document how to use mathtext to obtain unicode minus instead of dashes for negative numbers
When using `FormatStrFormatter` on the *x*-axis, the negative numbers have a dash - in front instead of a minus −.

```python
θ = linspace(-1.5, 0) * 2 * π
y = sin(θ)
plot(θ / π, y)
gca().xaxis.set_major_formatter(FormatStrFormatter('%g π'))
```

-sine-π

Without the formatter, all looks good.

```python
x = linspace(-1.5, 0)
y = sin(x * 2 * π)
plot(x, y)
```

-sine

I'm wondering who's handling the conversion of the dash into a minus, and how to recreate that when using `FormatStrFormatter`.

### PR DESCRIPTION
**Pull request by The-Inky**
## PR Summary
Amended docstring for FormatStrFormatter so there is documentation for how to get a unicode minus instead of a dash.

Closes #17967.

## PR Checklist

- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [x] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [x] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/matplotlib/ticker.py [hunk 1]
```diff
@@ -411,6 +411,10 @@ class FormatStrFormatter(Formatter):
 
     The format string should have a single variable format (%) in it.
     It will be applied to the value (not the position) of the tick.
+
+    Negative numeric values will use a dash not a unicode minus,
+    use mathtext to get a unicode minus by wrappping the format specifier
+    with $ (e.g. $%g$).
```
Comment (timhoffm) [inline lib/matplotlib/ticker.py:417]:
```suggestion
    with $ (e.g. "$%g$").
```
Let's mark the example explicitly as a string.

Comment (The-Inky) [reply lib/matplotlib/ticker.py]: Made the suggested change.

**Review by timhoffm — APPROVED** [review]

**Comment by timhoffm** [comment]
Thanks, and congratulations on your first contribution to Matplotlib! We hope to see you back.

**Comment by QuLogic** [comment]
@meeseeksdev backport to v3.4.x

**Pull request merged**